airflow/providers/imap/hooks/imap.py

```diff
@@ -71,10 +71,11 @@ def get_conn(self) -> 'ImapHook':
         """
         if not self.mail_client:
             conn = self.get_connection(self.imap_conn_id)
-            if conn.port:
-                self.mail_client = imaplib.IMAP4_SSL(conn.host, conn.port)
-            else:
-                self.mail_client = imaplib.IMAP4_SSL(conn.host)
+
+            if conn.extra_dejson.get('use_ssl', True):
+                self.mail_client = imaplib.IMAP4_SSL(conn.host, conn.port or imaplib.IMAP4_SSL_PORT)
+            else:  # fallback to standard imap connection
+                self.mail_client = imaplib.IMAP4(conn.host, conn.port or imaplib.IMAP4_PORT)
```

Comment:
See discussions in #20651 for context why we explicitly chose to _not_ implement this the way you’re doing it.

Reply:
Oh didn't know that. Thank you for pointing out! :)